clang/lib/CodeGen/BackendUtil.cpp

```diff
@@ -321,6 +322,41 @@ static bool actionRequiresCodeGen(BackendAction Action) {
          Action != Backend_EmitLL;
 }
 
+static std::string flattenClangCommandLine(ArrayRef<std::string> Args,
+                                      StringRef MainFilename) {
+  std::string FlatCmdLine;
+  raw_string_ostream OS(FlatCmdLine);
+  bool PrintedOneArg = false;
+  if (!StringRef(Args[0]).contains("-cc1")) {
+    llvm::sys::printArg(OS, "-cc1", /*Quote=*/true);
+    PrintedOneArg = true;
+  }
+  for (unsigned i = 0; i < Args.size(); i++) {
```

Comment:
Seems like this would be easier with a range based loop. However, skipping the argument would then need a flag. I'll let you decide which feels simpler.

Reply:
I think I'd keep it like this. I just moved it over and added some braces for the if conditions, leaving the original logic the same.

Reply:
Okay